tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -14,6 +14,18 @@ limitations under the License.
 ==============================================================================*/
 namespace tf_backend {
 
+
+/*==============================================================================
+
+  Please do not use RequestManager for new code.
+
+  We've generally found code that uses XMLHttpRequest without promises is
```

Comment:
Hmm, I'm not sure I see promises as being a problem - they're pretty standard these days and used in quite a bit of our codebase.  The new `fetch` API uses promises natively, and while I don't object to still using XMLHttpRequest if you like that API, I wouldn't want to rule out being able to use `fetch` alongside it as browser support permits.

Reply:
Code like this:

```js
// Source: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
fetch(url, {
  method: 'POST', // or 'PUT'
  body: JSON.stringify(data), // data can be `string` or {object}!
  headers: new Headers({
    'Content-Type': 'application/json'
  })
}).then(res => res.json())
.catch(error => console.error('Error:', error))
.then(response => console.log('Success:', response));
```

terrifies me. I would prefer to see less code like that.